### ISSUE TITLE
autofmt not respecting indent style for expressions

### ISSUE THREAD

**Issue by Enduriel** [DioxusLabs/dioxus#2537]
**Problem**
Autofmt is currently unable to fully support a hard tab indentation style for tabs, this is caused by the dependency on `prettyplease` which currently spits out strictly 4 spaced indentation.

**Steps to Reproduce**
Easiest method is to add a sample to the autofmt tests, for example
```
rsx! {
	div {
		onclick: |e| {
			let bla = 120;
			bla
		}
	}
}
```
or
```
rsx! {
[tab]div {
[tab][tab]onclick: |e| {
[tab][tab][tab]let bla = 120;
[tab][tab][tab]bla
[tab][tab]}
[tab]}
}
```

This test will fail as it will instead be formatted as
```
rsx! {
	div {
		onclick: |e| {
		    let bla = 120;
		    bla
		}
	}
}
```
or
```
rsx! {
[tab]div {
[tab][tab]onclick: |e| {
[tab][tab][4spaces]let bla = 120;
[tab][tab][4spaces]bla
[tab][tab]}
[tab]}
}
```

I've looked into fixing this on the side of Dioxus but it would require some pretty ugly hacks around [`unparse_expr`](https://github.com/DioxusLabs/dioxus/blob/487570d89751b34bbfd5e9b5ff1e0fd3780bf332/packages/autofmt/src/prettier_please.rs#L97), in addition to some special multiline string (and other edge cases?) parsing.

While this is technically possible but I'm not sure if you would be interested in this somewhat janky solution, but if so I would be willing to take a look at making it work.

I have also opened an [issue](https://github.com/dtolnay/prettyplease/issues/75) in the upstream `prettyplease` crate, and if that support gets added there that would be an alternative solution as well.

- [X] I'm interested in fixing this myself but don't know where to start

**Comment by Coder2195Text** [comment]
Related Issue
https://github.com/DioxusLabs/dioxus/issues/2524

**Comment by Enduriel** [comment]
Closed in favor of older issue #2524
